packages/embarkjs/embarkjs/src/lib/blockchain.js

```diff
@@ -208,15 +208,20 @@ Blockchain.doConnect = function(connectionList, opts, doneCb) {
   });
 };
 
+// See https://eips.ethereum.org/EIPS/eip-1102
 Blockchain.enableEthereum = function() {
   if (typeof window !== 'undefined' && window.ethereum) {
-    return ethereum.enable().then((accounts) => {
+    return ethereum.enable().then(() => {
       this.blockchainConnector.setProvider(ethereum);
-      this.blockchainConnector.setDefaultAccount(accounts[0]);
-      contracts.forEach(contract => {
-        contract.options.from = this.blockchainConnector.getDefaultAccount();
+
+      return this.blockchainConnector.getAccounts().then(accounts => {
+        this.blockchainConnector.setDefaultAccount(accounts[0]);
+        contracts.forEach(contract => {
+          contract.options.from = this.blockchainConnector.getDefaultAccount();
+        });
+        return accounts;
```

Comment:
This change just made me realize that our return of accounts never worked.
We return the accounts here at the end of the loop, but it's not received by anyone.
What we need to do is wrap the whole in a promise like so:
```
Blockchain.enableEthereum = function() {
  return new Promise((resolve, reject) => {
    if (typeof window === 'undefined' || !window.ethereum) {
      return reject(new Error('ethereum not available in this browser'));
    }
    ethereum.enable().then(() => {
      this.blockchainConnector.setProvider(ethereum);

      this.blockchainConnector.getAccounts().then(accounts => {
        this.blockchainConnector.setDefaultAccount(accounts[0]);
        contracts.forEach(contract => {
          contract.options.from = this.blockchainConnector.getDefaultAccount();
        });
        resolve(accounts);
      });
    }).catch(e => reject(e));
  });
};
```

Anyway, nothing wrong with your code, we just messed that up from the start.

Reply:
Could an `async function() {...}` be used instead? That way instead of `.then` / `.catch` it's possible to use `await`, i.e. `await this.blockchainConnector.getAccounts()`.

Reply:
Agree, it's even better